library/std/src/sys/xous/time.rs

```diff
@@ -0,0 +1,59 @@
+use crate::time::Duration;
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+pub struct Instant(Duration);
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+pub struct SystemTime(Duration);
+
+pub const UNIX_EPOCH: SystemTime = SystemTime(Duration::from_secs(0));
+
+impl Instant {
+    pub fn now() -> Instant {
+        let result = crate::os::xous::ffi::blocking_scalar(
+            crate::os::xous::services::ticktimer_server(),
+            [0 /* ElapsedMs */, 0, 0, 0, 0],
```

Comment:
is it worth making all the `ElapsedMs`-like things into an enum or named-consts somewhere?

or structs which have an `Into<[usize; 5]>` (or a different trait used as the `*_scalar()` argument) and appropriate constructors that wrap/document the other possible arguments?

Reply:
I do like that design idea. I initially made the `ffi` functions all `pub`, because it would be really nice to provide that functionality to all users. For example, I'd like to add it to `getrandom` in order to talk to the TRNG server. However, I made them `pub(crate)` in order to allow for suggestions such as this one.

I think the API arguments probably should go under `crate::os::xous::services`.

Reply:
The latest push uses this approach.